source/extensions/filters/network/thrift_proxy/router/router_ratelimit_impl.cc

```diff
@@ -102,6 +102,9 @@ RateLimitPolicyEntryImpl::RateLimitPolicyEntryImpl(
     case envoy::config::route::v3::RateLimit::Action::ActionSpecifierCase::kDestinationCluster:
       actions_.emplace_back(new DestinationClusterAction());
       break;
+    case envoy::config::route::v3::RateLimit::Action::ActionSpecifierCase::kQueryParameters:
+      throw EnvoyException("Query parameter rate limiting is not supported for Thrift");
+      break;
```

Comment:
even this is unnecessary because all unsupported cased will be handled at the `default` case.

Reply:
Make sense, I'll just remove it and the accompanying test cases.